common/src/main/java/org/apache/sedona/common/Functions.java

```diff
@@ -881,6 +881,38 @@ public static Integer nRings(Geometry geometry) throws Exception {
         return numRings;
     }
 
+    public static Geometry translate(Geometry geometry, double deltaX, double deltaY, double deltaZ) {
+        if (!geometry.isEmpty()) {
+            if (geometry instanceof GeometryCollection) {
+                GeometryCollection geometryCollection = (GeometryCollection) geometry;
```

Comment:
GeometryCollection can have child GeometryCollection. The process to extract GeometryCollection is more complicated than you thought. See here: https://github.com/apache/sedona/blob/master/common/src/main/java/org/apache/sedona/common/utils/GeomUtils.java#L411

Please fix it accordingly. We usually use a queue or recursive function to implement similar logics. In this case, I would recommend a recursive function. With such a function, the code could be even further simplified.

Reply:
On this regard, I think the approach used in ST_Force3D are actually better since it does not need to fetch individual geometries.

In addition, I noticed that ST_Force3D does not have test cases for GeomtryCollection/MultiPolygon and such. Would you please confirm ST_Force3D works for these cases as well?

Reply:
Fixed ST_Translate to handle recursive geometries in a better way.
Updated test cases for ST_Force3D